### DISCUSSION TITLE
Reenable Link-Time Optimization (LTO)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [romanz/electrs#1092] [Ideas]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

I see that LTO was introduced to the project in [this](https://github.com/romanz/electrs/commit/83bbfaaca105163dddf6fda96fcb367fdcb17ef5) commit but was removed [here](https://github.com/romanz/electrs/commit/3780d7d48bdc96db6b0df6f3eb955015e1779a60#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542). I wasn't able to find the reasons for that. Maybe it's a good time to enable LTO once again ;)

Thank you.

**Comment by romanz** [comment]
Good catch - thanks!
I've probably forgot to enable it when working on 0.9.0 release... could you please open a PR to enable it?

**Reply by zamazan4ik** [reply]
Sure: https://github.com/romanz/electrs/pull/1094
